test_util/helpers.py

```diff
@@ -31,6 +33,36 @@ def authenticate(self, cluster):
         logging.info('Authentication successful')
 
 
+def retry_boto_rate_limits(boto_fn, wait=2, timeout=60 * 60):
+    """Decorator to make boto functions resilient to AWS rate limiting and throttling.
+    If one of these errors is encounterd, the function will sleep for a geometrically
+    increasing amount of time
+    """
+    def ignore_rate_errors(*args, **kwargs):
+        local_wait = wait
+        local_timeout = timeout
+        while local_timeout > 0:
+            try:
+                return boto_fn(*args, **kwargs)
+            except (ClientError, WaiterError) as e:
+                if isinstance(ClientError, e):
+                    error_code = e.response['Error']['Code']
+                elif isinstance(WaiterError, e):
+                    error_code = e.last_response['Error']['Code']
+                else:
+                    raise
+                if error_code in ['Throttling', 'RequestLimitExceeded']:
+                    logging.warn('AWS API Limiting error: {}'.format(error_code))
+                    logging.warn('Sleeping for {} seconds before retrying'.format(local_wait))
+                    time.sleep(local_wait)
+                    local_timeout -= wait
```

Comment:
Yes, but I figured it was relatively insignificant in light of a geometric backoff. However, we can track time here if desired